### DISCUSSION TITLE
How do i update kafka config under spec.kafka.config

### DISCUSSION THREAD

**Discussion by superfly-git** [strimzi/strimzi-kafka-operator#4581] [Q&A]
How can i update an existing cluster configuration properties that are specified under spec.kafka.config .

For example i would like to update settings related to replication such as which now have value 1 specified in them.
default.replication.factor: 2
offsets.topic.replication.factor: 2
transaction.state.log.min.isr: 2
transaction.state.log.replication.factor: 2

From the kafka docs i can see that these values are read-only https://kafka.apache.org/documentation/#brokerconfigs_default.replication.factor

Does that mean i will need to destroy and recreate the cluster from scratch. One thing to be noted is that i use ephemeral storage for both the zookeeper and the braker nodes so any recreation of the cluster will require recreation of the topics as well.

**Comment by scholzj** [comment]
You just edit the Kafka custom resource to update them. Either using `kubectl edit ...` or by updating your YAML file and doing `kubectl apply ...`. And the operator takes care of the rest.

**Reply by superfly-git** [reply]
that doesn look to be working. i have updated the yaml definition and reapplied using kubectl apply -f def.yaml but it does not show the updated conf values in the kafka CRD object

**Reply by scholzj** [reply]
Then you probably did it wrong or missed some error you got from the `kubectl apply`. Or you have some other tools in your cluster which rewrite it back (such as GitOps etc.)

**Reply by superfly-git** [reply]
you are correct i missed something. Once i applied again i see that the pods where recreated. If i was using ebs storage for the brokers the messages where going to be available after the recreation of the pods(i know noob question here)?

**Reply by scholzj** [reply]
Yes, with persistent storage, it would just continue where it left of after the restart.
